### ISSUE TITLE
Aerial Companion corrupt file on updates

### ISSUE THREAD

**Issue by lmkunert** [JohnCoates/Aerial#1078]
#### General troubleshooting tips

Before logging an issue please check that:
- [ ] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [ ] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [ ] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [ ] Mac model: MBP 15" 2017
- [ ] macOS version: 10.15.6
- [ ] Monitor setup: external via TB3

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

Using Aerial Companion (1.0.1) every single time I try to update the screensaver I get an error that says the file is corrupted. I'm having to manually update the file myself still rendering the companion app moot for me. Any guidance?
![image](https://user-images.githubusercontent.com/3722319/91202914-16e25200-e6c8-11ea-85fa-e8c34bc60e2c.png)

**Comment by glouel** [comment]
Hi @lmkunert 

Are you running some kind of firewall possibly (like Little Snitch), or a company proxy, or something of the kind? This is intriguing... sorry about the bug in any case let's see if we can troubleshoot it. Thanks!

**Comment by lmkunert** [comment]
No firewall enabled on my machine nor using anything like Little Snitch, but it is a work computer that I'm having the issue on so perhaps some kind of weird proxy issue. I've also noticed I have to use Safari to download the newest version manually, otherwise it says the file is corrupted when downloaded via Chrome. Perhaps that's just life for me now on this work computer. :(

**Comment by lmkunert** [comment]
PS: I also have to use the xattr -cr trick on the .saver file even when I download from Safari.

**Comment by glouel** [comment]
@lmkunert you probably didn't see my edit: one thing you could do possibly, go to ~/Library/Application Support/AerialUpdater and look for a Aerial.saver.zip in there. If you see it, can you show me it's file size, etc ?

This file may be super small and just be a webpage from your company proxy, just a guess.

Regarding chrome yes that's an issue with the zip distribution sadly, it actually says the file is "uncommon" if I remember correctly, this gets fixed after a few hours. The zip is still notarized with apple.

**Comment by lmkunert** [comment]
Looks like the proper file size of 6.6 MB.

**Comment by glouel** [comment]
Could you enable the Debug Log in companion ? it's in About:

<img width="453" alt="Capture d’écran 2020-08-25 à 18 58 18" src="https://user-images.githubusercontent.com/37544189/91204856-f1c00480-e704-11ea-860b-cb84855de35c.png">

Also could you run 

shasum -a 256 Aerial.saver.zip 

in that directory ?

**Comment by glouel** [comment]
For 2.0.4 this is what you should get : 
$ shasum -a 256 Aerial.saver.zip
9212164ce0cb63d1aae0d39552d18de1acc745e6b1c35878611e8d08ab2fa336  Aerial.saver.zip

**Comment by lmkunert** [comment]
Unless this is an older version, I'm not getting a matching checksum:
ed1bf2e840db93f435beabd569d65b85e6a5618e58f8e39788616d2a6e4bd15c Aerial.saver.zip

**Comment by glouel** [comment]
Could you try deleting the file just to see and try again maybe ? Is it possible your company proxy has a anti virus that checks the file in the zip and somehow rezip them (maybe with a txt file inside or something?). Kinda crazy but that may be that, hmm...

**Comment by lmkunert** [comment]
Since I've manually updated it already it isn't giving the option to download w/ companion app after I restarted it, so will try again if there's another dot release. Thanks for the help. Very weird one for sure.

**Comment by glouel** [comment]
Please keep me posted, there might not be a new version tonight but there will be tomorrow worst case. In the meantime if you feel like it, one thing you could try is just right click the screen saver in the screen saver list and press delete, then you can try to reinstall. Your settings will remain.

**Comment by lmkunert** [comment]
Ah, great idea. I went ahead and did the delete and got another checksum entirely again, so there must be something like you said with my corporate network stuff modifying it. I actually see the .saver file inside the zip file, though.

2a28553bdec77d21facd3d024569c7f852570c83a4b3e8961b90a31a2f7e332c Aerial.saver.zip

**Comment by glouel** [comment]
That's so weird... Any chance you can attach that file here (heads up, you can't attach by mail you have to do it on github directly) ? Would love to have a look and see what can be done. If it's only my signed file rezipped, guess I could add some option to disable the zip checksum check for users like you who would need it, it's a bit weird though!

**Comment by glouel** [comment]
So, I tried a few things, and this is so weird...

My original distribution : 
`181 files, 18337754 bytes uncompressed, 6524431 bytes compressed:  64.4%`

Your file : 
`181 files, 18337754 bytes uncompressed, 6524437 bytes compressed:  64.4%`

If I try to md5 the dirs, unzipped, though, they don't match.

What I've noticed is that your corporate thing might be removing some links that were put there by Xcode : 

```
finishing deferred symbolic links:
  Aerial.saver/Contents/Frameworks/OAuthSwift.framework/Resources -> Versions/Current/Resources
  Aerial.saver/Contents/Frameworks/OAuthSwift.framework/Versions/Current -> A
  Aerial.saver/Contents/Frameworks/OAuthSwift.framework/OAuthSwift -> Versions/Current/OAuthSwift
```

My theory, the corporate thing does unzip on a windows machine or somethingn and something gets lost on the rezip. 

Here are the links on my distribution : 

```
$ ls -la
total 0
drwxr-xr-x   5 glouel  staff  160 25 aoû 12:11 .
drwxr-xr-x  30 glouel  staff  960 25 aoû 12:11 ..
lrwxr-xr-x   1 glouel  staff   27 25 aoû 20:26 OAuthSwift -> Versions/Current/OAuthSwift
lrwxr-xr-x   1 glouel  staff   26 25 aoû 20:26 Resources -> Versions/Current/Resources
drwxr-xr-x   4 glouel  staff  128 25 aoû 12:11 Versions
```

And the missing links in your file:

```
$ ls -la
total 16
drwxrwxr--@  5 glouel  staff  160 25 aoû 12:11 .
drwxrwxr--@ 30 glouel  staff  960 25 aoû 12:11 ..
-rw-rw-r--@  1 glouel  staff   27 25 aoû 14:11 OAuthSwift
-rw-rw-r--@  1 glouel  staff   26 25 aoû 14:11 Resources
drwxrwxr--@  4 glouel  staff  128 25 aoû 12:11 Versions
```

So... I guess it's kinda corrupted ? I'm so puzzled about why one would design something that would do that, that I'm not sure what to say. You may not be able to install it either through homebrew for the exact same reason (cause they are pretty much doing the same thing for that specific part), or have other issues with other packages that happen to be zipped. 

Quite frankly I'm not sure what to say or do here. I didn't try your .saver locally, FYI OAuthSwift is what is used by Weather, so maybe weather is broken with that file, who knows. Have I said it's weird already?

**Comment by lmkunert** [comment]
Definitely one of the weirdest things I've seen no question. Thanks for digging into it. Locked down corporate machines do weird stuff I suppose lol. I guess we can close this one as a "WTF can't really fix it" type situation. At least I know how to manage it for now anyways. 🙂  Thanks again.

**Comment by glouel** [comment]
I have to agree, you probably win the weird issue of the year right here and now, one of the wildest thing I've seen !

I'd be curious if the weather overlay works or not (you can also just try the test thing in the overlays ui) considering it's that library that's missing stuff. Weirdly enough macOS codesigning didn't care at all, but I guess they don't sign symbolic links. I guess if it does work, maybe I could add an option to remove the sha check for corporate environments with a big large warning or something.

**Comment by lmkunert** [comment]
The weather overlay works for me when I manually install the normal .saver file that I have downloaded. I haven't tried to install the 'corrupted' saver file, though, if that's what you mean.

**Comment by glouel** [comment]
(just wanted to say, you are not alone, someone chimed in on discord with the exact same issue :D)

I'll make an adjustment to the error message, mentioning corporate environments.

**Comment by ZV00** [comment]
Gone ahead and implemented this into the FAQ so that others can learn more about the issue. https://aerialscreensaver.github.io/faq.html#InstallionErrors

**Comment by glouel** [comment]
CLosing this as we now have proper updates via Sparkle.
